src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs

```diff
@@ -206,18 +206,18 @@ public static async ValueTask<QuicConnection> ConnectQuicAsync(string host, int
 
             if (lastException != null)
             {
-                throw CreateWrappedException(lastException, cancellationToken);
+                throw CreateWrappedException(lastException, host, port, cancellationToken);
             }
 
             // TODO: find correct exception to throw here.
             throw new HttpRequestException("No host found.");
         }
 
-        private static Exception CreateWrappedException(Exception error, CancellationToken cancellationToken)
+        private static Exception CreateWrappedException(Exception error, string host, int port, CancellationToken cancellationToken)
         {
             return CancellationHelper.ShouldWrapInOperationCanceledException(error, cancellationToken) ?
                 CancellationHelper.CreateOperationCanceledException(error, cancellationToken) :
-                new HttpRequestException(error.Message, error, RequestRetryType.RetryOnNextProxy);
+                new HttpRequestException($"{error.Message} {host}:{port}", error, RequestRetryType.RetryOnNextProxy);
```

Comment:
How this would look if `host` is IPv6 address? Do you know if it already comes in with the surrounding `[]`?

Reply:
Yes, the brackets are being added here:
https://github.com/dotnet/runtime/blob/4b03513e20e0caf9ee34817eb74356903f1bb33e/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpAuthority.cs#L26-L29